From 2bcae0199842035b99ae28659f821b51fbc9b537 Mon Sep 17 00:00:00 2001
From: Nicolas Boulenguez <nicolas.boulenguez@free.fr>
Date: Wed, 3 Apr 2024 17:43:50 +0200
Subject: [PATCH 8/9] Ada: import nanosleep from
 System.OS_Primitives.Timed_Delay

It improves readability to import this C function in the file where it
is actually used.
---
 gcc/ada/libgnat/s-optide.adb            | 7 +++++++
 gcc/ada/libgnat/s-osprim__posix.adb     | 9 ---------
 gcc/ada/libgnat/s-osprim__posix2008.adb | 8 --------
 3 files changed, 7 insertions(+), 17 deletions(-)

diff --git a/src/gcc/ada/libgnat/s-optide.adb b/src/gcc/ada/libgnat/s-optide.adb
index e15493c..4ad16a2 100644
--- a/src/gcc/ada/libgnat/s-optide.adb
+++ b/src/gcc/ada/libgnat/s-optide.adb
@@ -36,6 +36,13 @@ procedure Timed_Delay
   (Time : Duration;
    Mode : Integer)
 is
+
+   function nanosleep (rqtp, rmtp : not null access C_Time.timespec)
+                      return Integer
+     with Import,
+          Convention => C,
+          External_Name => "nanosleep";
+
    Request    : aliased C_Time.timespec;
    Remaind    : aliased C_Time.timespec;
    Rel_Time   : Duration;
diff --git a/src/gcc/ada/libgnat/s-osprim__posix.adb b/src/gcc/ada/libgnat/s-osprim__posix.adb
index bb9a28d..b331d5c 100644
--- a/src/gcc/ada/libgnat/s-osprim__posix.adb
+++ b/src/gcc/ada/libgnat/s-osprim__posix.adb
@@ -34,15 +34,6 @@ with System.C_Time;
 
 package body System.OS_Primitives is
 
-   --  ??? These definitions are duplicated from System.OS_Interface
-   --  because we don't want to depend on any package. Consider removing
-   --  these declarations in System.OS_Interface and move these ones in
-   --  the spec.
-
-   function nanosleep (rqtp, rmtp : not null access C_Time.timespec)
-                      return Integer;
-   pragma Import (C, nanosleep, "nanosleep");
-
    -----------
    -- Clock --
    -----------
diff --git a/src/gcc/ada/libgnat/s-osprim__posix2008.adb b/src/gcc/ada/libgnat/s-osprim__posix2008.adb
index 377fb3a..7f44efa 100644
--- a/src/gcc/ada/libgnat/s-osprim__posix2008.adb
+++ b/src/gcc/ada/libgnat/s-osprim__posix2008.adb
@@ -39,14 +39,6 @@ package body System.OS_Primitives is
 
    subtype int is System.CRTL.int;
 
-   --  ??? These definitions are duplicated from System.OS_Interface because
-   --  we don't want to depend on any package. Consider removing these
-   --  declarations in System.OS_Interface and move these ones to the spec.
-
-   function nanosleep (rqtp, rmtp : not null access C_Time.timespec)
-                      return Integer;
-   pragma Import (C, nanosleep, "nanosleep");
-
    -----------
    -- Clock --
    -----------
-- 
2.39.2

